amr-wind/equation_systems/icns/source_terms/GeostrophicForcing.cpp

```diff
@@ -30,26 +31,35 @@ GeostrophicForcing::GeostrophicForcing(const CFDSim& /*unused*/)
         amrex::ParmParse pp("CoriolisForcing");
         amrex::Real rot_time_period = 86400.0;
         pp.query("rotational_time_period", rot_time_period);
+        // Latitude is mandatory, everything else is optional
+        // Latitude is read in degrees
+        pp.get("latitude", latitude);
         coriolis_factor = 2.0 * utils::two_pi() / rot_time_period;
         amrex::Print() << "Geostrophic forcing: Coriolis factor = "
                        << coriolis_factor << std::endl;
-
-        amrex::Real latitude = 90.0;
-        pp.query("latitude", latitude);
-        AMREX_ALWAYS_ASSERT(
-            amrex::Math::abs(latitude - 90.0) <
-            static_cast<amrex::Real>(vs::DTraits<float>::eps()));
     }
 
     {
         // Read the geostrophic wind speed vector (in m/s)
         amrex::ParmParse pp("GeostrophicForcing");
         pp.getarr("geostrophic_wind", m_target_vel);
+        latitude = utils::radians(latitude);
+        sinphi = std::sin(latitude);
+        cosphi = std::cos(latitude);
     }
 
+    {
+        amrex::ParmParse pp("ABL");
+        pp.query("three_ComponentForcing", m_three_dimensional_forcing);
+    }
+
+    amrex::Real S = (m_three_dimensional_forcing == true) ? 1.0 : 0.0;
+
     m_g_forcing = {
-        -coriolis_factor * m_target_vel[1], coriolis_factor * m_target_vel[0],
-        0.0};
+        -coriolis_factor * m_target_vel[1] * sinphi +
+            coriolis_factor * m_target_vel[3] * cosphi * S,
```

Comment:
this is causing the unit test to fail since m_target_vel is length 3 so you can only use indices 0,1,2. This probably means the unit tests are not sufficient either if you didn't catch this before